src/components/BondingCurve.vue

```diff
@@ -0,0 +1,158 @@
+<script>
+import { Scatter } from 'vue-chartjs';
+
+export default {
+  name: 'BondingCurve',
+  extends: Scatter,
+  props: {
+    data: {
+      type: Object,
+      default: null,
+    },
+  },
+  mounted() {
+    this.renderChart(this.chartData, this.chartOptions);
+  },
+  computed: {
+    chartData() {
+      const sellPrice = this.data.sellPrice / 1000000000000000000;
+      const buyPrice = this.data.buyPrice / 1000000000000000000;
+      const supply = this.data.totalSupply / 1000000000000000000;
+      const borderDash = [4];
+      return {
+        showLines: true,
+        datasets: [
+          {
+            order: 6,
+            fill: '+1',
+            data: [
+              { x: 0, y: 0 },
+              { x: supply, y: buyPrice },
+            ],
+            backgroundColor: '#1161fe3f',
+          },
+          {
+            order: 7,
+            fill: true,
+            data: [
+              { x: 0, y: 0 },
+              { x: supply, y: sellPrice },
+            ],
+            backgroundColor: '#6224c75f',
+          },
+          {
+            order: 3,
+            data: [
+              { x: supply, y: buyPrice },
+              { x: supply, y: 0 },
+            ],
+            borderColor: '#1161fe',
+            borderDash,
+          },
+          {
+            order: 4,
+            data: [
+              { x: 0, y: 0 },
+              { x: 3 * supply, y: 3 * sellPrice },
```

Comment:
where does 3 come from? what if a different bonding curve contract is used?

Reply:
we would need to get some prediction for the future to display, that is hinted in the design (~3 times the supply). Since it is a linear curve that starts from 0, it is a simple multiplication.
Is there a way to get appropriate data?

Reply:
You can fetch prices straight from the contract. Given a total supply and amount for buy/sell you can get the price.